### PR TITLE
Getting ready for RC1

### LINKED ISSUE #1832: NonEmptyVector.last
`NonEmptyVector` could have a total `last` method (unlike `Vector`). I think it would be useful.

(`NonEmptyList` could also have this, but I don't know how useful would it be, since it would need to walk the whole list...)

### PR DESCRIPTION
**Pull request by kailuowang**
RC1 is the only planned RC release. 1.0.0 would probably be within a month after (unless we found more bugs that neccessets 

Outstanding items (ones currently scheduled with RC1)

## PRs

Most of them only 1 maintainer approval short. 

- [x] Add SortedMap instances #1972  
- [x] Add a few type classes to generated tuple instances (continued)   #1967
- [x] Add CommutativeApply and CommutativeApplicative   #1927  
- [x] Add Invariant instances for kernel type classes #1914
- [x] Sync NEL and NEV methods (fixes #1832)   #1838
- [x] Add Parallel type class   #1837
- [x] Welcome, Alleycats #1984

## Issues

- [x] Foldable.toList potentially inconsistent for Map #1831
  - [x] Move traverse, foldable instance for Map and Set to alleycats, add laws (no ticket yet, pending #1984)
- [x] Foldable.iterateRight is not sound API  #1973 
- [ ] Kliesli is not stack-safe on left-associated binds #1733
- ~[ ] ??? Consider adding some form newtypes to cats #1800~

??? If we can't reach consensus on it, we probably have to release without it. 

## Nice to have

- [x] Reconsider the use of fold in Validated #1951 

## Please add here if you notice any missing items.  
Update: removed  #1800 from RC1 scope

### CODE REVIEW

**Comment by iravid** [comment]
Would be good to include https://github.com/typelevel/cats/issues/1733 in this list. `Sync` for `ReaderT[IO, E, ?]` seems very important.

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,163 @@
...
+* [#1934](https://github.com/typelevel/cats/pull/1934) Restruct `functor`  by @kailuowang
+* [#1803](https://github.com/typelevel/cats/pull/1803) Convert `ReaderWriterStateT` to `IndexedReaderWriterStateT`  by @iravid
+* [#1775](https://github.com/typelevel/cats/pull/1775) Convert `StateT` to `IndexedStateT`  by @iravid
+* [#1098](https://github.com/typelevel/cats/pull/1098) Add a different `MonoidK` and `SemigroupK` instance for `Kleisli`  by @peterneyens
+* [#1922](https://github.com/typelevel/cats/pull/#922) Make kernel laws consistent with core laws  by @LukaJCB
+
+To migrate from 1.0.0-MF.
+
+* The rename of `Cartesian` to `Semigroupal` and `EitherT.liftT` to `EitherT.liftF` can be done automatically through scalafix we provide. See instructions [here](https://github.com/typelevel/cats/blob/master/scalafix/README.md)
+* For `FlatMap`'s `>>` and `<<`, use `Apply`'s `>*` and `<*` instead
```
Comment (iravid) [inline CHANGES.md]: Typo - `>*` instead of `*>`

**Comment by kailuowang** [comment]
@iravid for #1733, @johnynek 's solution seems good, does it necessitate an API addition to in cats (vs. letting users hand roll themselves)? If so, would you propose the new API in the issue?

**Comment by iravid** [comment]
I unfortunately cannot comment intelligently on the proposed solution (other than that it would necessitate a `cats-core -> cats-free` dependency) as I don't understand how the usage of flatMap should look.

**Comment by ceedubs** [comment]
> Reconsider the use of fold in Validated #1951

Since this is a change that could be made in a completely compatible way, I don't see much of a reason that a 1.0 release should be blocked by it.

**Comment by kailuowang** [comment]
@ceedubs that's true, I had the same thought, I included it because the performance penalty seems significant and the fix is simple (if no-one else, I can give it an attack this week), I am going to move it out of the list though.

**Comment by LukaJCB** [comment]
What about #1831? I think we should address that in some way maybe

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,185 @@
...
+* [#1955](https://github.com/typelevel/cats/pull/1955) Deprecate `FlatMap`'s `>>` and `<<`  by @LukaJCB
+* [#1947](https://github.com/typelevel/cats/pull/1947) Rename `EitherT.liftT` to `EitherT.liftF`  by @aeons
+* [#1934](https://github.com/typelevel/cats/pull/1934) Restruct `functor`  by @kailuowang
+* [#1803](https://github.com/typelevel/cats/pull/1803) Convert `ReaderWriterStateT` to `IndexedReaderWriterStateT`  by @iravid
+* [#1775](https://github.com/typelevel/cats/pull/1775) Convert `StateT` to `IndexedStateT`  by @iravid
+* [#1098](https://github.com/typelevel/cats/pull/1098) Add a different `MonoidK` and `SemigroupK` instance for `Kleisli`  by @peterneyens
+* [#1922](https://github.com/typelevel/cats/pull/1922) Make kernel laws consistent with core laws  by @LukaJCB
+* [#1838](https://github.com/typelevel/cats/pull/1838) Sync `NonEmptyList` and `NonEmptyVector` methods by @durban
+* [#1914](https://github.com/typelevel/cats/pull/1914) Add `Invariant` instances for kernel type classes by @LukaJCB 
+* [#1980](https://github.com/typelevel/cats/pull/1980) Make `iterateRight` in `Foldable` sound  by @LukaJCB
```
Comment (johnynek) [inline CHANGES.md]: this is also a bug fix. I would either list it twice or move it there (I doubt any user will be hit by it as an incompatibility).

Comment (LukaJCB) [reply CHANGES.md]: My PR approval still stands if you decide to add this @kailuowang :) So go ahead and merge afterwards 🎉

**Pull request merged**